cloudfoundry-client-reactor/src/main/java/org/cloudfoundry/reactor/tokenprovider/AbstractUaaTokenProvider.java

```diff
@@ -105,7 +105,33 @@ public String getClientSecret() {
 
     @Override
     public final Mono<String> getToken(ConnectionContext connectionContext) {
-        return this.accessTokens.computeIfAbsent(connectionContext, this::token);
+        Mono<String> accessToken = this.accessTokens.get(connectionContext);
+        if(accessToken != null) {
+        	try {
+        	String token = this.accessTokens.get(connectionContext).map(s -> s.split(" ")[1]).block();
```

Comment:
Making a `.block()` call is a significant code smell in a reactive project (note that, outside of a few specific tests, it's not used anywhere else in this project). This needs to be refactored into a reactive style.